Спринт 13: Экран Аудиоплеер

john-krasinski opened the pull request.

Commit:
Спринт 13: Экран Аудиоплеер

IliaAnK commented:
Экран настройки:

В качестве заголовка экрана лучше использовать `Toolbar`. Это стандартный компонент. В него можно передать текст заголовка, иконку(стрелочка влево) с помощью атрибута `app:navigationIcon` и обработать нажатие с помощью `setNavigationOnClickListener {}`. Характеристики текста можно задать через стиль и атрибута `titleTextAppearance`.
Для остальных экранов **аналогично**.

Рекомендация к каждой конструкции
```
<FrameLayout> 
    <TextView/>
    <ImageView/> 
</FrameLayout>
```
Во-первых, это излишняя вложенность, которая негативно сказывается при отрисовке фрейма.
Во-вторых, она вызывает трудности с понимание того как свёрстан экран;
В-третьих, проще формировать и применять стили.

Каждая такая конструкция может быть представлена одной View.
Для TextView с иконкой справа:
```
<com.google.android.material.textview.MaterialTextView
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:drawableEnd="@drawable/share"
    android:paddingHorizontal="16dp"
    android:paddingVertical="20dp"
    android:text="@string/shareAppSetting"
    android:textSize="16sp" />
```

Общие характеристика вынести само собой можно вынести в стили.

Итого иерархия всего `layout`-а будет вот такой:
```
<LinearLayout>
    <MaterialToolbar/>
    <SwitchMaterial/>
    <MaterialTextView/>
    <MaterialTextView/>
    <MaterialTextView/>
<LinearLayout/>
```

john-krasinski commented:
К сожалению, пока что в курсе все примеры были с конструкцией вида
```xml
<FrameLayout> 
    <TextView/>
    <ImageView/> 
</FrameLayout>
```
В различные тонкости верстки пока не вдавались, постараюсь исправить.
Я хотел оставить такую структуру, потому что так можно поставить обработчик OnClick только на картинку, а иначе при нажатии на текст заголовока тоже будет закрываться активити. Есть какой-то способ реализовать такое поведение или не стоит заморачиваться?

Upd.
Посмотрел код, такая структура используется только в поле поиска для скрытия кнопки очистки. А заголовки уже используют TextView с иконкой слева, раньше делали замечания и я поправил это, только обернуты в LinearLayout, потому что не обновлены экраны с учетом новых знаний о ConstraintLayout.

Буду разбираться с Toolbar, спасибо

Commit:
Спринт 13: фиксы

закругление углов обложки
выравнивание названия трека
цвет фона
баг с переключением темы
